apps/search/js/providers/grid_provider.js

```diff
@@ -21,8 +23,15 @@ DataGridProvider.prototype = {
 
   render: function(results) {
     results.forEach(function(config, index) {
+      if (this.grid.getLength() >= this.MAX_RESULTS) {
```

Comment:
Since this is a workaround, I would prefer to avoid modifying gaia-grid at all, we can keep a count of how many results we have displayed in the grid provider right?